extensions/dacpac/src/test/testContext.ts

```diff
@@ -113,9 +113,16 @@ export function createViewContext(): ViewTestContext {
 		onDidClick: onClick.event
 	});
 
-	let buttonBuilder: azdata.ComponentBuilder<azdata.ButtonComponent> = {
+	let buttonBuilder: azdata.ComponentBuilder<azdata.ButtonComponent, azdata.ButtonProperties> = {
 		component: () => button,
-		withProperties: (properties) => {
+		withProperties: (properties: any) => {
+			if ((properties as any).label === '•••') {
+				button.label = '•••';
+				button.onDidClick = fileButtonOnClick.event;
+			}
+			return buttonBuilder;
+		},
+		withProps: (properties) => {
```

Comment:
just for confirmation the explicit setting like component.property = value will continue to work as before.. yeah?

Reply:
yes, this is not changing any existing behavior, just adding a withProps() method that should make life easier.